### DISCUSSION TITLE
Trying and trying but i cant get it to work.

### DISCUSSION THREAD

**Discussion by Zulgeteb** [ORelio/Minecraft-Console-Client#2423] [General]
Hello everyone,
i have the following code in my .ini.
So when i connect it logs on nice.
It selects worldtype on server nice.
I connect to my afk-spot and start auto selling.
The server restarts 03:00 and everyone is kicked to lobby for 1-2 minutes.
During this time i dont want to autosell items.

[code]
[ChatBot.ScriptScheduler]
Enabled = true

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Login"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = false, Times = [ 23:59:59, ] }
Trigger_On_Interval = { Enable = false, MinTime = 120.0, MaxTime = 120.0 }
Action = "send /login *****PASS*****"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Select World"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = true, Times = [ 03:03:00, ] }
Trigger_On_Interval = { Enable = false, MinTime = 1.0, MaxTime = 2.0 }
Action = "script select.txt"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sell All"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = false, Times = [ 23:59:59, ] }
Trigger_On_Interval = { Enable = true, MinTime = 1.2, MaxTime = 1.2 }
Action = "send /sell all"
[/code]

Now i wanted to use /execif and %sales% so i can manually do /set sales=on (or true/1/false/off/0 (whatever works))
And i want taskscheduler to turn off sales 30 sec before server restarts and turn it on again 30 sec after world is selected.
So i changed it to:

[code]
[ChatBot.ScriptScheduler]
Enabled = true

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Login"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = false, Times = [ 23:59:59, ] }
Trigger_On_Interval = { Enable = false, MinTime = 120.0, MaxTime = 120.0 }
Action = "send /login *****PASS*****"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Select World"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = true, Times = [ 03:03:00, ] }
Trigger_On_Interval = { Enable = false, MinTime = 1.0, MaxTime = 2.0 }
Action = "script select.txt"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sales On"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = true, Times = [ 03:03:30, ] }
Trigger_On_Interval = { Enable = false, MinTime = 3600.0, MaxTime = 3600.0 }
Action = "/set sales = on"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sales Off"
Trigger_On_First_Login = false
Trigger_On_Login = false
Trigger_On_Times = { Enable = true, Times = [ 02:59:30, ] }
Trigger_On_Interval = { Enable = false, MinTime = 3600.0, MaxTime = 3600.0 }
Action = "/set sales = off"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sell All"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = false, Times = [ 23:59:59, ] }
Trigger_On_Interval = { Enable = true, MinTime = 1.2, MaxTime = 1.2 }
Action = "execif %sales% == "on" --> send /sell all"
[/code]

Tried several ways with and without quotes/slashes but i cant get it to work.

**Comment by Zulgeteb** [comment]
Ok maybe i have to break it down.
Is this ok --> Action = "/set sales = on" ,
and this --> Action = "/set sales = off" ,
to set value of %sales% ?
If there wasnt an error in above then there is definitely one in-->  Action = "execif %sales% == "on" --> send /sell all"

**Reply by ReinforceZwei** [reply]
I think the `%` around the variable is not needed, so it should be `execif sales == "on" --> send /sell all`.

**Reply by Zulgeteb** [reply]
OK so last part of ini is now:

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sales On"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = true, Times = [ 03:03:30, ] }
Trigger_On_Interval = { Enable = false, MinTime = 3600.0, MaxTime = 3600.0 }
Action = "/set sales = on"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sales Off"
Trigger_On_First_Login = false
Trigger_On_Login = false
Trigger_On_Times = { Enable = true, Times = [ 02:59:30, ] }
Trigger_On_Interval = { Enable = false, MinTime = 3600.0, MaxTime = 3600.0 }
Action = "/set sales = off"

[[ChatBot.ScriptScheduler.TaskList]]
Task_Name = "Sell All"
Trigger_On_First_Login = false
Trigger_On_Login = true
Trigger_On_Times = { Enable = false, Times = [ 23:59:59, ] }
Trigger_On_Interval = { Enable = true, MinTime = 1.2, MaxTime = 1.2 }
Action = execif sales == "on" --> send /sell all 

It comes with error about line after last Action.

Failed to load settings:
Expected the start of a number, string literal, boolean, array, or table on line 483, found 'e'
I get similar error when i put a / in front of execif.

I think the execif part has to be in quotes (but that messes with the "on").

**Reply by Zulgeteb** [reply]
And is there a way yo read out 'sales' (on/off) value when MCC is running?
So i can check if the /set sales = off/on works. (assuming i can use '/set sales = on/off' like i can do '/script select.txt'.

**Reply by ReinforceZwei** [reply]
Try this: `Action = "execif sales == \"on\" --> send /sell all"`

**Reply by Zulgeteb** [reply]
I am getting somewhere now.
Finally had MCC loading but now i had an in app error, so i added / in front of execif then MCC changed the line into Action =  '/execif sales == "on" --> send /sell all' 
Now it does load and doesnt show any in app errors but it doesnt sell as expected, even sales is set to on when it starts up.

**Reply by Zulgeteb** [reply]
Also figured out from https://mccteam.github.io/guide/usage.html#execif its not --> but ---> (3 dashes instead of 2), but still doesn't work.
No idea what i am doing wrong.
Also is there a way to read out 'sales' so i can maybe figure out what is going wrong (maybe /set sales is setup wrong?).

Action = "send /sell all" , does work (is what i am using so far but needs my attention at 3:00 (something i'd like to skip))
Action = "execif sales == \"on\" ---> send /sell all" is what would like to get working (MCC changed it to Action = 'execif sales == "on" ---> send /sell all'
It might not work incase: Action = "/set sales = off" (or on), isn't correct.
